src/test/java/de/tum/in/www1/artemis/iris/PyrisLectureIngestionTest.java

```diff
@@ -0,0 +1,104 @@
+import de.tum.in.www1.artemis.service.connectors.pyris.PyrisWebhookService;
+import de.tum.in.www1.artemis.user.UserUtilService;
+
+class PyrisLectureIngestionTest extends AbstractIrisIntegrationTest {
+
+    private static final String TEST_PREFIX = "pyrislectureingestiontest";
+
+    @Autowired
+    private PyrisWebhookService pyrisWebhookService;
+
+    @Autowired
+    private CourseRepository courseRepository;
+
+    @Autowired
+    private LectureRepository lectureRepository;
+
+    @Autowired
+    private UserUtilService userUtilService;
+
+    @Autowired
+    private CourseUtilService courseUtilService;
+
+    @Autowired
+    private LectureUtilService lectureUtilService;
+
+    @Autowired
+    private PyrisJobService jobService;
+
+    private Lecture lecture1;
+
+    @BeforeEach
+    void initTestCase() throws Exception {
+        for (String job : jobService.getAllJobs().keySet()) {
+            jobService.removeJob(job);
+        }
+        userUtilService.addUsers(TEST_PREFIX, 1, 1, 0, 1);
+        List<Course> courses = courseUtilService.createCoursesWithExercisesAndLectures(TEST_PREFIX, true, true, 1);
+        Course course1 = this.courseRepository.findByIdWithExercisesAndExerciseDetailsAndLecturesElseThrow(courses.getFirst().getId());
+        this.lecture1 = course1.getLectures().stream().findFirst().orElseThrow();
+        this.lecture1.setTitle("Lecture " + lecture1.getId()); // needed for search by title
+        this.lecture1 = lectureRepository.save(this.lecture1);
+        // Add users that are not in the course
+        userUtilService.createAndSaveUser(TEST_PREFIX + "student42");
+        userUtilService.createAndSaveUser(TEST_PREFIX + "tutor42");
+        userUtilService.createAndSaveUser(TEST_PREFIX + "instructor42");
+
+        int numberOfSlides = 2;
+        AttachmentUnit attachmentUnitWithSlides = lectureUtilService.createAttachmentUnitWithSlidesAndFile(numberOfSlides);
+        lecture1 = lectureUtilService.addLectureUnitsToLecture(lecture1, List.of(attachmentUnitWithSlides));
+        this.lecture1 = lectureRepository.findByIdWithLectureUnitsAndAttachmentsElseThrow(lecture1.getId());
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testPyrisLectureIngestionWebhookJob_isEmpty_When_subsettingsDisabled() throws Exception {
+        Course course2 = courseUtilService.addEmptyCourse();
+        courseUtilService.enableMessagingForCourse(course2);
+        request.postWithResponseBody("/api/lectures/import/" + lecture1.getId() + "?courseId=" + course2.getId(), null, Lecture.class, HttpStatus.CREATED);
+        if (lecture1.getLectureUnits().getFirst() instanceof AttachmentUnit attachmentUnit) {
+            pyrisWebhookService.executeLectureIngestionPipeline(true, List.of(attachmentUnit));
+        }
+        Set<String> keySet = jobService.getAllJobs().keySet();
+        assertThat(keySet).isEmpty();
```

Comment:
This is a pretty weak check if the pipeline was run or not. Ideally you would use the existing IrisRequestMockProvider to check that no call was done to Pyris.
Also this is not safe when running the entire test suite, as other tests may also add keys, which may not be removed. Then this check would erroneously fail.